### ISSUE TITLE
SUBMIT AND RUN TEST BUTTON DISSAPEARED

### ISSUE THREAD

**Issue by izzygardnerr** [exercism/exercism#6446]
Hi, 

I completed an exercise which resulted in an error.

However, after making changes the run tests and submit buttons have completely disappeared.

I don’t know if it’s an error on my behalf or a bug on Excercism.  It’s really annoying as I cannot complete any other exercises now.

**Comment by kytrinyx** [comment]
This sounds like a bug, and very frustrating! Would you please take a screen shot of what you're seeing so that we can look for clues?

**Comment by izzygardnerr** [comment]
Hey!

Here is a screenshot - I cannot scroll beyond this point so I cannot see
the 'run tests' or 'submit' buttons.

Thank you :)

Best,

Izzy

**Comment by kotp** [comment]
I did not see what was attached, but I wonder if you reduce the font size, or drag the comment window to resize it if it reappears.

**Comment by kytrinyx** [comment]
@izzygardnerr It looks like email attachments in replies don't get included in the issue comment. Would you try visiting this thread and uploading the screenshot directly?

**Comment by izzygardnerr** [comment]
> @izzygardnerr It looks like email attachments in replies don't get included in the issue comment. Would you try visiting this thread and uploading the screenshot directly?

<img width="1440" alt="Screenshot 2022-07-19 at 21 01 42" src="https://user-images.githubusercontent.com/103532518/179931922-49a25bff-f1dc-4390-ba17-abb35d5dc97d.png">

**Comment by izzygardnerr** [comment]
@kytrinyx From this point I cannot scroll down and if I make changes to the code (i.e. add the notes) which was the final step in the exercise, I am still unable to scroll down to see the run test or submit buttons.  There is not a way for me to reduce font size, however I have zoomed out on the page and it has still not resolved the issue.

**Comment by izzygardnerr** [comment]
@iHiD Hey - it's Safari.

The version details are: Version 14.0.3 (16610.4.3.1.4)

I have ad-block, that's it.

Best,

Izzy

**Comment by izzygardnerr** [comment]
@iHiD @kytrinyx 

Hi,

I removed my browser extension and there has not been any changes unfortunately.

Do you have any further suggestions?

**Comment by kytrinyx** [comment]
We'll try to do some testing to see if we can reproduce it.

**Comment by izzygardnerr** [comment]
@kytrinyx hey, sorry for the late reply.

I clicked this - the code has reverted to the beginning.

However, the same issues that the run test and submits button do not appear remains.

**Comment by kytrinyx** [comment]
You tried disabling the ad-block, right? It sounded like it from the discussion above, but I want to be sure.

**Comment by izzygardnerr** [comment]
Hi,

Yes I have:)

**Comment by izzygardnerr** [comment]
Hi Jeremy,

- It is just this one.  However, I have only done one previous exercise
before this.  The other exercises are locked for me so I'm unable to test
this.
- Yes, the buttons were present, I had actually passed 3/4 tests and was in
the process of adding notes to my code as the final test.
- I will try the different browser and get back to you.

Best,

Izzy

**Comment by izzygardnerr** [comment]
@SleeplessByte thank you.

What do you mean by 100vh?

Please let me know.

Best,

Izzy

**Comment by kytrinyx** [comment]
@izzygardnerr 100vh refers to a technical thing in CSS3. You don't need to know the details unless you're actually curious. (To me it sounds like @SleeplessByte was saying that maybe we could switch to 100vh in the CSS, which might fix it for you).

Here's an article about it: https://medium.com/rbi-tech/safaris-100vh-problem-3412e6f13716

**Comment by SleeplessByte** [comment]
Yeah @kytrinyx is right. Sorry @izzygardnerr, the comment was directed at @iHiD.

I'm having a hard time reproducing the bug, but I do know Safari _has_ had issues with the specific markup Exercism is using. 

If you feel like it, @izzygardnerr, there is something you could try to help us help you. No worries if this doesn't work for you, or this is hard, but it can help us narrow down the possible issues.

1. Make any change in your solution. White space change (such as a space character) is fine, even on an empty line.
2. Turn the editor into _accessibility_ mode. You do this by clicking **Settings**, visually indicated by the gear, and then selecting _accessibility_ under **tab mode**. 
   ![Pop up of settings on Exercism Editor](https://user-images.githubusercontent.com/1964376/181276543-e50e9279-0550-4956-97a5-02569903912c.png)
3. After clicking it, press <kbd>Tab</kbd>. Visually, a box should appear around **Open more options** (the three dots). This means that the _focus_ (where your keyboard location is) has moved.
   ![Focus is on the next button, open more options](https://user-images.githubusercontent.com/1964376/181276850-4680684f-6a29-467f-9741-b7365cad9a35.png)
4. Press <kbd>Tab</kbd> again. `lasagna.py` should now be selected.
5. Press <kbd>Tab</kbd> again. The entire editor should now be selected. Take a screenshot for us. Why? We want to see if the box ends at the bottom or extends past it.
6. Press <kbd>Tab</kbd> again. You should _not_ see any boxes now.
7. Press <kbd>Tab</kbd> again. If the buttons are somewhere on the page, it has _now_ selected `Run Tests`. Press <kbd>Enter</kbd>. Let us know if the tests ran?

-----

Sitenote for @iHiD , at point 6, the following element is selected but no focus is visible:

```html
<div
  spellcheck="false"
  autocorrect="off"
  autocapitalize="off"
  translate="no"
  contenteditable="true"
  class="cm-content cm-lineWrapping focus-visible"
  style="tab-size: 4;"
  role="textbox"
  aria-multiline="true"
  aria-autocomplete="list"
  aria-expanded="false"
  data-focus-visible-added=""
>
...
</div>
```

This is the _correct_ element to focus, but the element at point 5 is "inert". No key press will do anything. Probably something you want to check out.

**Comment by izzygardnerr** [comment]
Hi,

Here is a screenshot of when I tabbed and the box became highlighted.

I accidentally pressed revert to exercise start - not sure if this will
affect anything but I did not see the run tests and submit buttons after
following the steps you mentioned.

I installed Google Chrome however and the buttons are there now for the
time being.

I will attempt the exercise there and see if it works the whole way through.

Best,

Izzy

**Comment by izzygardnerr** [comment]
Sorry for the late reply btw!

**Comment by SleeplessByte** [comment]
@iHiD Izzy indicating it works in chrome and not Firefox makes it more likely that this is a position bug in Safari that likely has since been fixed, but of course is present for them as Safari updates are tied to the OS.
